tests/foreman/ui/test_architecture.py

```diff
@@ -32,14 +34,17 @@ def test_positive_create_arch_with_os(self, test_data):
         @Assert: Architecture is created
 
         """
-        entities.OperatingSystem(name=test_data['os_name']).create()
         with Session(self.browser) as session:
-            make_arch(session, name=test_data['name'],
-                      os_names=[test_data['os_name']])
-            self.assertIsNotNone(self.architecture.search(test_data['name']))
-
-    @data(*generate_strings_list())
-    def test_positive_create_arch_with_different_names(self, name):
+            for test_data in valid_arch_os_names():
+                with self.subTest(test_data):
+                    entities.OperatingSystem(
+                        name=test_data['os_name']).create()
+                    make_arch(session, name=test_data['name'],
```

Comment:
Also note that the following code is commented too:
```py
{u'name': gen_string('html'), u'os_name': gen_string('html')},
```
This passed many times in the local run because of fauxfactory issue/behavior: https://github.com/omaciel/fauxfactory/issues/80.  This will fail when the supplied html is more than 32 characters in length.

IMO my code changes gives a reliable set of tests (ofcourse without html tests) which will add great value to Satellite. When the UI framework is fixed to accept two string values, html strings can be opened up.